syntax = "proto3";

package room;

service Room {
  rpc getRoomIdentifier (RoomIdentifierRequest) returns (RoomIdentifierResponse) {}
  rpc getRoomCapacity (RoomCapacityRequest) returns (RoomCapacityResponse) {}
  rpc getAllRoomsInOffice (OfficeName) returns (RoomsInOffice){}
}

message RoomIdentifierRequest{
  string office = 1;
  string roomName = 2;
}

message RoomIdentifierResponse{
  string roomIdentifier = 1;
}

message RoomCapacityRequest{
  string roomIdentifier = 1;
}

message RoomCapacityResponse{
  int32 capacity = 1;
}

message OfficeName{
  string officeName = 1;
}

message RoomsInOffice{
  repeated AllRoomsInOffice rooms = 1;
}

message AllRoomsInOffice{
  string name = 1;
  string capacity = 2;
  string office = 3;
  bool whiteboard = 4;
  bool projector = 5;
}
